babel/support.py

```diff
@@ -71,18 +69,25 @@ def date(self, date: _date | None = None, format: _PredefinedTimeFormat | str =
         """
         return format_date(date, format, locale=self.locale)
 
-    def datetime(self, datetime: _date | None = None, format: _PredefinedTimeFormat | str = 'medium') -> str:
+    def datetime(
+        self,
+        datetime: datetime.date | datetime.datetime | None = None,
```

Comment:
Doesn't really need both date and datetime since datetime is a subclass of date
```py
>>> issubclass(datetime.datetime, datetime.date)
True
```
```suggestion
        datetime: datetime.date | None = None,
```